internal/provider/serverless_cluster.go

```diff
@@ -0,0 +1,168 @@
+			ClusterName:          d.Get("name").(string),
+			BroadcastType:        "PUBLIC",
+			CidrBlock:            "172.31.0.0/16",
+			CloudProviderID:      1,
+			AccountCredentialID:  1,
+			RegionID:             1,
+			ReplicationFactor:    3,
+			NumberOfNodes:        3,
+			UserAPIInterface:     "CQL",
+			InstanceID:           74,
+			FreeTier:             true,
+			EnableDNSAssociation: true,
+			Provisioning:         "serverless",
+		}
+	)
+
+	cr, err := c.CreateCluster(ctx, r)
+	if err != nil {
+		return diag.Errorf("error creating serverless cluster: %w", err)
+	}
+
+	if err := waitForCluster(ctx, c, cr.ID); err != nil {
+		return diag.Errorf("error waiting for serverless cluster: %w", err)
+	}
+
+	cluster, err := c.GetCluster(ctx, cr.ClusterID)
+	if err != nil {
+		return diag.Errorf("error reading serverless cluster: %w", err)
+	}
+
+	d.SetId(strconv.Itoa(int(cr.ClusterID)))
+	d.Set("free_tier", cluster.FreeTier)
+
+	return nil
+}
+
+func resourceServerlessClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		c = meta.(*scylla.Client)
+	)
+
+	clusterID, err := strconv.ParseInt(d.Id(), 10, 64)
+	if err != nil {
+		return diag.Errorf("error reading id=%q: %w", d.Id(), err)
+	}
+
+	reqs, err := c.ListClusterRequest(ctx, clusterID, "CREATE_CLUSTER")
+	if err != nil {
+		return diag.Errorf("error reading serverless cluster request: %w", err)
+	}
+	if len(reqs) != 1 {
+		return diag.Errorf("unexpected number of serverless cluster requests, expected 1, got: %+v", reqs)
+	}
+
+	if reqs[0].Status != "COMPLETED" {
+		if err := waitForCluster(ctx, c, reqs[0].ID); err != nil {
+			return diag.Errorf("error waiting for serverless cluster: %w", err)
+		}
+	}
+
+	cluster, err := c.GetCluster(ctx, clusterID)
+	if err != nil {
+		return diag.Errorf("error reading serverless cluster: %w", err)
+	}
+
+	d.SetId(strconv.Itoa(int(clusterID)))
+	d.Set("free_tier", cluster.FreeTier)
+
+	return nil
+}
+
+func resourceServerlessClusterUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// Scylla Cloud API does not support updating a serverless cluster,
+	// thus the update always fails
+	return diag.Errorf(`updating "scylladbcloud_serverless_cluster" resource is not supported`)
+}
+
+func resourceServerlessClusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		c = meta.(*scylla.Client)
+	)
+
+	clusterID, err := strconv.ParseInt(d.Id(), 10, 64)
+	if err != nil {
+		return diag.Errorf("error reading id=%q: %w", d.Id(), err)
+	}
+
+	name, ok := d.GetOk("name")
+	if !ok {
+		return diag.Errorf("unable to read serverless cluster name from state file")
+	}
+
+	r, err := c.DeleteCluster(ctx, clusterID, name.(string))
+	if err != nil {
+		return diag.Errorf("error deleting serverlessCluster: %w", err)
+	}
+
+	if !strings.EqualFold(r.Status, "QUEUED") && !strings.EqualFold(r.Status, "IN_PROGRESS") {
```

Comment:
something like "DELETED" or "DONE" (don't remember which final status it uses) also not fine?

Reply:
it's `"COMPLETED"`, I added it both to `cluster` and `serverless_cluster`